src/components/views/rooms/EntityTile.js

```diff
@@ -92,17 +92,31 @@ const EntityTile = React.createClass({
         };
     },
 
+    componentWillMount: function() {
+        // We use a listener on the document so we can find out when the cursor leaves the element more
+        // safely. This is a workaround for https://github.com/facebook/react/issues/4492 and
+        // https://github.com/vector-im/riot-web/issues/1997
+        document.addEventListener("mouseover", this.mouseEnter);
+    },
+
+    componentWillUnmount: function() {
+        document.removeEventListener("mouseover", this.mouseEnter);
+    },
+
     shouldComponentUpdate: function(nextProps, nextState) {
         if (this.state.hover !== nextState.hover) return true;
         return this.props.shouldComponentUpdate(nextProps, nextState);
     },
 
-    mouseEnter: function(e) {
-        this.setState({ 'hover': true });
-    },
+    mouseEnter: function(ev) {
+        let targetHover = false;
+        if (this.container && (this.container === ev.target || this.container.contains(ev.target))) {
```

Comment:
Hmmm, having this called for every DOM element you enter and leave in the whole app feels suboptimal indeed. I'm curious what didn't work well with the CSS approach, did you try something like this?:

```
.mx_EntityTile .mx_PresenceLabel {
	display: none;
}

.mx_EntityTile:not(.mx_EntityTile_surpressOnHover):hover .mx_PresenceLabel {
	display: block;
}
```

(and set the `mx_EntityTile_surpressOnHover` class when state.suppressOnHover is true). If this would work, I think this might allow us to simplify the render method and css as well for this component.

Reply:
When I tried it before it just wasn't working whatsoever. Not sure what I did wrong, but it seems to work fine now. Have switched to CSS.